types/got/index.d.ts

```diff
@@ -90,11 +90,13 @@ declare namespace got {
 
     interface GotJSONFn {
         (url: GotUrl): GotPromise<any>;
+        (url: GotUrl, options: GotOptions<string | null>): GotPromise<any>;
         (url: GotUrl, options: GotJSONOptions): GotPromise<any>;
```

Comment:
Looks like overload with `GotJSONOptions` is redundant?

Reply:
Nevermind.

I think correct type here is `Partial<GotJSONOptions>`.

Reply:
Yeah, that's a bit smarter 😄